src/core/carousel/components/CarouselSliderDot.tsx

```diff
@@ -4,19 +4,57 @@ import classNames from 'classnames';
 import styles from '../carousel.module.scss';
 import { getSlideDotKey } from '../utils/utils';
 import { useCarouselContext } from '../context/CarouselContext';
+import { useTranslationWithFallback } from '../../translation/useTranslationWithFallback';
+
+function CarouselSlideDot({ slideIndex }: { slideIndex: number }) {
+  const { currentSlide, handleUpdateSlideProps, navigateWithDots } =
+    useCarouselContext();
+
+  const { t } = useTranslationWithFallback();
+
+  const onClickHandler = (slideNumber: number): void => {
+    if (slideNumber === currentSlide) {
+      return;
+    }
+    if (slideNumber > currentSlide) {
+      handleUpdateSlideProps(slideNumber);
+    } else {
+      handleUpdateSlideProps(-slideNumber);
```

Comment:
Why is the slideNumber made negative here? Isn't all the necessary information already in the slideNumber itself without changing its sign?